### PR TITLE
rethrow more exceptions when finalizing a ws connection

### PR DESCRIPTION
**Pull request by amesgen**
When closing the WS connection in the resource finalizer there might be an exception. Maybe there is an additional, helpful exception stuck in the queue (received in `onError` in the JDK WS listener), so we rethrow it. So this change improves diagnostics in some rare cases.

Initially discussed in #116

### CODE REVIEW

File: core/src/main/scala/org/http4s/client/jdkhttpclient/JdkWSClient.scala [hunk 1]
```diff
@@ -83,13 +87,23 @@ object JdkWSClient {
...
               for {
                 isOutputOpen <- F.delay(!webSocket.isOutputClosed)
                 closeOutput = fromCompletableFuture(
                   F.delay(webSocket.sendClose(JWebSocket.NORMAL_CLOSURE, ""))
                 )
-                _ <- closeOutput.whenA(isOutputOpen)
+                _ <- closeOutput.whenA(isOutputOpen).onError {
+                  case e: IOException =>
+                    for {
+                      chunk <- queue.tryDequeueChunk1(10)
```
Comment (amesgen) [inline core/src/main/scala/org/http4s/client/jdkhttpclient/JdkWSClient.scala]: This `10` is arbitrary, `1` should be enough, but I think it doesn't hurt.

**Comment by amesgen** [comment]
travis failure is unrelated..............

**Comment by rossabaker** [comment]
Goodness, those 403s are irritating.  I wonder whether GitHub Actions suffers as much?

**Comment by amesgen** [comment]
I am actually motivated to try to migrate this repo to GH Actions CI (prbbly tomorrow).

**Comment by rossabaker** [comment]
In addition to being an excellent module in its own right, this repo has been a nice proving ground for other improvements.  Would love to see that tried here.

**Pull request merged**